src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -51,15 +60,34 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
      */
     @PostMapping(value = Constants.PROGRAMMING_SUBMISSION_RESOURCE_PATH + "{participationId}")
     public ResponseEntity<?> notifyPush(@PathVariable("participationId") Long participationId, @RequestBody Object requestBody) {
-
         log.info("REST request to inform about new commit+push for participation: {}", participationId);
-        programmingSubmissionService.notifyPush(participationId, requestBody);
+
+        try {
+            ProgrammingSubmission submission = programmingSubmissionService.notifyPush(participationId, requestBody);
+            // notify the user via websocket.
+            messagingTemplate.convertAndSend("/topic/participation/" + participationId + "/newSubmission", submission);
+        }
+        catch (IllegalArgumentException ex) {
+            log.error("Exception encountered when trying to extract the commit hash from the request body. Processing submission for participation {} with request object {}: {}",
+                    participationId, requestBody, ex);
+            return badRequest();
+        }
+        catch (IllegalStateException ex) {
+            log.error("Tried to create another submission for the same commitHash and participation! Processing submission for participation {} with request object {}: {}",
+                    participationId, requestBody, ex);
+            return badRequest();
+        }
+        catch (EntityNotFoundException ex) {
+            log.error("Participation with id {} is not a ProgrammingExerciseParticipation! Processing submission for participation {} with request object {}: {}", participationId,
```

Comment:
I would prefer 

> Processing submission for participation {} **failed** with request object {}: {}

Reply:
Why do you want to make the error logging more generic instead of giving the indication about what happened most likely?